### PR TITLE
GH-239 add --strict <true|false> support for add-command

### PR DESCRIPTION
**Pull request by eedwardsav**
### overview

This PR addresses #239 by adding support for a `--strict <true|false>` option when using `stubbs: add-command`

This results in a new property in a command's `metadata` file called `STRICT=` being added when using `add-command`.

The value defaults to `true` to preserve pre-existing behavior.

`generate-options` will use this to determine whether to add the `help option` (see below)

Disabling 'strict argument checking' prevents the following block of code from being added to `options.sh`:
```
            # help option
            -|--*?) echo >&2 "unrecognized option: $OPT"
                rerun_option_usage
                exit 2
                ;;
```
By not adding this into options.sh, commands that are not recognized as 'rerun commands' by rerun are allowed to be passed onto `_CMD_LINE`, which addresses #239

### review needed

The area of interest that I believe needs to be looked at is the changes to `generate-options`, as I interpreted the `bash-pedantic` version to be an attempt at being more POSIX compliant (?), but I was unable to confirm. Thus, the changes should work in both versions, but may need adjustment depending on the intent of the `pedantic` version.

Also note that `-template` appeared to be a previously supported option for `add-command`, but is no longer, so the freshly generated `options.sh` for `add-command` reflects this.

### test notes

Tests are passing locally -- I updated `add-command-1-test.sh` to reflect `--strict`

### squashed commits
GH-239 explicit exit code due to moving metadata write operation
GH-239 moved order of creating metadata since generate options now has dependency on it
GH-239 added logic for disabling strict checking
GH-239 add support for --strict <true> to add-command

### CODE REVIEW

**Comment by stagrlee** [comment]
Took a quick peak at this.  Rerun should parses the command options, but you have code that does that.  Seems odd.  I'll try to look some more tomorrow.

**Comment by eedwardsav** [comment]
@stagrlee No worries we can sync on it when you're free -- likely the code you're seeing is the code that was auto-generated by rerun

File: modules/stubbs/commands/add-command/script [hunk 1]
```diff
@@ -100,6 +105,10 @@ done
     read DESCRIPTION
 }
 
+if [ -z "${STRICT:-}" ]; then
```
Comment (stagrlee) [inline modules/stubbs/commands/add-command/script]: This needs to move down the file into the implementation section

**Comment by eedwardsav** [comment]
Closing PR in favor of a simpler approach described by stagrlee in #239

**Pull request closed without merging**